src/core/Akka/Actor/Deploy.cs

```diff
@@ -1,148 +1,223 @@
+        public Deploy(string path, Config config, RouterConfig routerConfig, Scope scope, string dispatcher,
+            string mailbox)
             : this()
         {
-            Path = path;
-            Config = config;
-            RouterConfig = routerConfig;
-            Scope = scope ?? NoScopeGiven;
-            Dispatcher = dispatcher ?? NoDispatcherGiven;
-            Mailbox = mailbox ?? NoMailboxGiven;
+            _path = path;
+            _config = config;
+            _routerConfig = routerConfig;
+            _scope = scope ?? NoScopeGiven;
+            _dispatcher = dispatcher ?? NoDispatcherGiven;
+            _mailbox = mailbox ?? NoMailboxGiven;
         }
 
-        public string Path { get; private set; }
-        public Config Config { get; private set; }
-        public RouterConfig RouterConfig { get; private set; }
-        public Scope Scope { get; private set; }
-        public string Mailbox { get; private set; }
-        public string Dispatcher { get; private set; }
+        public string Path
+        {
+            get { return _path; }
+        }
 
-        public Deploy WithFallback(Deploy other)
+        public Config Config
         {
-            return new Deploy
-            {
-                Path = Path,
-                Config = Config.WithFallback(other.Config),
-                RouterConfig = RouterConfig.WithFallback(other.RouterConfig),
-                Scope = Scope.WithFallback(other.Scope),
-                Dispatcher = Dispatcher == NoDispatcherGiven ? other.Dispatcher : Dispatcher,
-                Mailbox = Mailbox == NoMailboxGiven ? other.Mailbox : Mailbox,
-            };
+            get { return _config; }
         }
 
-        public Deploy Copy(Scope scope = null)
+        public RouterConfig RouterConfig
         {
-            return new Deploy
+            get { return _routerConfig; }
+        }
+
+        public Scope Scope
+        {
+            get { return _scope; }
+        }
+
+        public string Mailbox
+        {
+            get { return _mailbox; }
+        }
+
+        public string Dispatcher
+        {
+            get { return _dispatcher; }
+        }
+
+        public bool Equals(Deploy other)
+        {
+            if (other == null) return false;
+            return ((string.IsNullOrEmpty(Mailbox) && string.IsNullOrEmpty(other.Mailbox)) ||
+                    string.Equals(Mailbox, other.Mailbox)) &&
+                   string.Equals(Dispatcher, other.Dispatcher) &&
+                   string.Equals(Path, other.Path) &&
+                   RouterConfig.Equals(other.RouterConfig) &&
+                   ((Config.IsNullOrEmpty() && other.Config.IsNullOrEmpty()) ||
+                    Config.ToString().Equals(other.Config.ToString())) &&
+                   (Scope == null && other.Scope == null || (Scope != null && Scope.Equals(other.Scope)));
+        }
```

Comment:
Maybe switch to using the fields instead of getting the values via the properties, e.g. 
`string.Equals(_mailbox, other._mailbox)`